File: 백준/Silver/25206. 너의 평점은/너의 평점은.py
sum_grade = 0
total = 0
for i in range(20):
  dept, grade, score = input().split()
  grade = float(grade)

  if score == "A+":
    total += grade * 4.5
    sum_grade += grade

  elif score == "A0":
    total += grade * 4.0
    sum_grade += grade

  elif score == "B+":
    total += grade * 3.5
    sum_grade += grade

  elif score == "B0":
    total += grade * 3.0
    sum_grade += grade

  elif score == "C+":
    total += grade * 2.5
    sum_grade += grade

  elif score == "C0":
    total += grade * 2.0
    sum_grade += grade

  elif score == "D+":
    total += grade * 1.5
    sum_grade += grade

  elif score == "D0":
    total += grade * 1.0
    sum_grade += grade

  elif score == "F":
    total += grade * 0
    sum_grade += grade

print(total / sum_grade)

File: 백준/Silver/25206. 너의 평점은/너의 평점은.c
#include <stdio.h>
#include <string.h>

int main(void) {  
   float score;
   char dept[51], grade[3];
   
   float total = 0;
   float sum_score = 0;
   for(int i = 0; i < 20; i++) {
      scanf("%s %f %s", dept, &score, grade);
      if(strcmp(grade, "A+") == 0){
         total += score * 4.5;
      }
      else if(strcmp(grade, "A0") == 0){
         total += score * 4.0;
      }
      else if(strcmp(grade, "B+") == 0){
         total += score * 3.5;
      }
      else if(strcmp(grade, "B0") == 0){
         total += score * 3.0;
      }
      else if(strcmp(grade, "C+") == 0){
         total += score * 2.5;
      }
      else if(strcmp(grade, "C0") == 0){
         total += score * 2.0;
      }
      else if(strcmp(grade, "D+") == 0){
         total += score * 1.5;
      }
      else if(strcmp(grade, "D0") == 0){
         total += score * 1.0;
      }
      else if(strcmp(grade, "F") == 0){
         total += score * 0.0;
      }
      if(strcmp(grade, "P") != 0){
         sum_score += score;
      }
   }

   printf("%f", total / sum_score);
   
   return 0;
}